brave/brave-core
# Fix user earns while native Windows and macOS notifications are suppressed

Fixes https://github.com/brave/brave-browser/issues/3651

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Confirm native Ad notifications are not shown on Windows if Focus Assist is enabled
- Confirm native Ad notifications are not shown on Windows if notifications are disabled for the user
- Confirm native Ad notifications are not shown on Windows if notifications are disabled for application
- Confirm native Ad notifications are not shown on Windows if notifications are disabled by group policy
- Confirm native Ad notifications are not shown on Windows if notifications are disabled by manifest
- Confirm native Ad notifications are not shown on macOS 10.15 if notifications set to none
- Confirm native Ad notifications are not shown on macOS 10.14 and older if notifications set to none
- Confirm ads are shown if native notifications are set to banner or alert

Note: Notifications will always be shown for Chromium Message Center (Brave fallback if native notifications are not supported)

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
